src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -852,56 +852,67 @@ public void parseSetValueAction(ActionController source, Element e) {
         source.registerEventListener(event, action);
     }
 
+    public void parseSendAction(ActionController source, Element e) {
+        String event = this.getRequiredAttribute(e, "event");
+        String id = this.getRequiredAttribute(e, "submission");
+
+        SendAction action = new SendAction(id);
+        source.registerEventListener(event, action);
+    }
+
+    private String getRequiredAttribute(Element e, String attrName) {
+        String value = e.getAttributeValue(null, attrName);
+        if(value == null || value == "") {
+            throw new XFormParseException("Missing required attribute "+ attrName + " in element",
+                    e);
+        }
+        return value;
+    }
+
     private void parseSubmission(Element submission) {
         String id = submission.getAttributeValue(null, ID_ATTR);
 
-        //These two are always required
+        String resource = getRequiredAttribute(submission, "resource");
+        String targetref = getRequiredAttribute(submission, "targetref");
+
+        String ref = submission.getAttributeValue(null, "ref");
+
+        //For Validation Only
         String method = submission.getAttributeValue(null, "method");
-        String action = submission.getAttributeValue(null, "action");
 
-        SubmissionParser parser = new SubmissionParser();
-        for (SubmissionParser p : submissionParsers) {
-            if (p.matchesCustomMethod(method)) {
-                parser = p;
-            }
+        if(!("get".equals(method))) {
+            throw new XFormParseException("Unsupported submission @method: " + method);
         }
 
-        //These two might exist, but if neither do, we just assume you want the entire instance.
-        String ref = submission.getAttributeValue(null, REF_ATTR);
-        String bind = submission.getAttributeValue(null, BIND_ATTR);
+        String replace = submission.getAttributeValue(null, "replace");
 
-        XPathReference dataRef = null;
-        boolean refFromBind = false;
+        if(!("text".equals(replace))) {

```

Comment:
Why do we have these params at all if we're enforcing one value?

Reply:
This is an XForms specification, we're just implementing it. (Our XForms would be invalid if thy didn't define the parameter). This also protects us against accidentally parsing something the form would expect us to understand that we do not (IE: If we got a form from another platform with mode=POST, the validator should ensure that we fail and explain that we can't support it)